### PR TITLE
Feature setting context

### PR DESCRIPTION
**Pull request by marcinel123**
Hi Adam,

Inside this PR you will find my solution regarding fetching data from API and putting it inside React Context.

Please have a look.

Martin

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -49,6 +49,10 @@
 		"prettier/prettier": 0,
 		"jsx-a11y/label-has-associated-control": 0,
 		"react/jsx-props-no-spreading": 0,
-		"import/no-cycle": 0
+		"import/no-cycle": 0,
```
Comment (AdamKniec) [inline .eslintrc.json]: what's that?

File: .eslintrc.json [hunk 2]
```diff
@@ -49,6 +49,10 @@
...
+		"react-hooks/rules-of-hooks": 0,
```
Comment (AdamKniec) [inline .eslintrc.json]: what's that?

File: .eslintrc.json [hunk 3]
```diff
@@ -49,6 +49,10 @@
...
+		"no-unused-expressions": 0,
+		"@typescript-eslint/no-unused-expressions": 0,
```
Comment (AdamKniec) [inline .eslintrc.json]: Why did you turn this off?

Comment (marcinel123) [reply .eslintrc.json]: fixed

File: .eslintrc.json (hunk 2 above)
Comment (marcinel123) [reply .eslintrc.json]: fixed

File: .eslintrc.json (hunk 1 above)
Comment (marcinel123) [reply .eslintrc.json]: fixed

**Pull request merged**